[CI] RequestIndexFilteringIT testIndicesDontExist failing

**Build Scans:**
- [elasticsearch-intake #16426 / 8.18.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/p5jzmwpwpkr5k)
- [elasticsearch-intake #16425 / 8.18.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/vgw7vewhwyc7i)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-clusters:v8.18.0#newToOld" -Dtests.class="org.elasticsearch.xpack.esql.ccq.RequestIndexFilteringIT" -Dtests.method="testIndicesDontExist" -Dtests.seed=C00C10F899D5D5C1 -Dtests.bwc=true -Dtests.locale=raj -Dtests.timezone=Atlantic/Bermuda -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.ccq.RequestIndexFilteringIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testIndicesDontExist),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string containing "verification_exception"
     but: was "method [POST], host [http://[::1]:46477], URI [/_query?format=smile&pretty=true&error_trace=true], status line [HTTP/1.1 400 Bad Request]
:)
úerrorúroot_causeøútypePparsing_exceptionreasonàline 1:28: invalid index pattern [*:test1], remote clusters are not supported in LOOKUP JOINüstack_traceäorg.elasticsearch.xpack.esql.parser.ParsingException: line 1:28: invalid index pattern [*:test1], remote clusters are not supported in LOOKUP JOIN
	at org.elasticsearch.xpack.esql.parser.LogicalPlanBuilder.lambda$visitJoinCommand$19(LogicalPlanBuilder.java:594)
	at org.elasticsearch.xpack.esql.core.tree.Node.lambda$forEachUp$3(Node.java:92)
	at org.elasticsearch.xpack.esql.core.tree.Node.forEachUp(Node.java:85)
	at org.elasticsearch.xpack.esql.core.tree.Node.forEachUp(Node.java:90)
	at org.elasticsearch.xpack.esql.parser.LogicalPlanBuilder.lambda$visitJoinCommand$20(LogicalPlanBuilder.java:588)
	
[truncated]
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in step 8.18.0_bwc-snapshots
- [8.x] 2 failures in test testIndicesDontExist (0.8% fail rate in 248 executions)
- [8.x] 2 failures in step 8.18.0_bwc-snapshots (14.3% fail rate in 14 executions)
- [8.x] 2 failures in pipeline elasticsearch-intake (33.3% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -245,9 +245,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/120810
 - class: org.elasticsearch.indices.mapping.UpdateMappingIntegrationIT
   issue: https://github.com/elastic/elasticsearch/issues/116126
-- class: org.elasticsearch.xpack.esql.ccq.RequestIndexFilteringIT
-  method: testIndicesDontExist
-  issue: https://github.com/elastic/elasticsearch/issues/120889
 - class: org.elasticsearch.datastreams.DataStreamsClientYamlTestSuiteIT
   method: test {p0=data_stream/140_data_stream_aliases/Create data stream aliases using wildcard expression}
   issue: https://github.com/elastic/elasticsearch/issues/120890

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RequestIndexFilteringTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RequestIndexFilteringTestCase.java
@@ -13,6 +13,7 @@ import org.elasticsearch.client.Response;
 import org.elasticsearch.client.ResponseException;
 import org.elasticsearch.client.RestClient;
 import org.elasticsearch.test.rest.ESRestTestCase;
+import org.elasticsearch.transport.RemoteClusterAware;
 import org.elasticsearch.xcontent.XContentType;
 import org.elasticsearch.xpack.esql.AssertWarnings;
 import org.elasticsearch.xpack.esql.action.EsqlCapabilities;
@@ -212,13 +213,18 @@ public abstract class RequestIndexFilteringTestCase extends ESRestTestCase {
         assertThat(e.getMessage(), anyOf(containsString("no such index [foo]"), containsString("no such index [remote_cluster:foo]")));
 
         if (EsqlCapabilities.Cap.JOIN_LOOKUP_V12.isEnabled()) {
+            var pattern = from("test1");
             e = expectThrows(
                 ResponseException.class,
-                () -> runEsql(timestampFilter("gte", "2020-01-01").query(from("test1") + " | LOOKUP JOIN foo ON id1"))
+                () -> runEsql(timestampFilter("gte", "2020-01-01").query(pattern + " | LOOKUP JOIN foo ON id1"))
             );
             assertEquals(400, e.getResponse().getStatusLine().getStatusCode());
-            assertThat(e.getMessage(), containsString("verification_exception"));
-            assertThat(e.getMessage(), containsString("Unknown index [foo]"));
+            assertThat(
+                e.getMessage(),
+                RemoteClusterAware.isRemoteIndexName(pattern)
+                    ? allOf(containsString("parsing_exception"), containsString("remote clusters are not supported in LOOKUP JOIN"))
+                    : allOf(containsString("verification_exception"), containsString("Unknown index [foo]"))
+            );
         }
     }
 

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RequestIndexFilteringTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RequestIndexFilteringTestCase.java
@@ -221,6 +221,8 @@ public abstract class RequestIndexFilteringTestCase extends ESRestTestCase {
             assertEquals(400, e.getResponse().getStatusLine().getStatusCode());
             assertThat(
                 e.getMessage(),
+                // currently we don't support remote clusters in LOOKUP JOIN
+                // this check happens before resolving actual indices and results in a different error message
                 RemoteClusterAware.isRemoteIndexName(pattern)
                     ? allOf(containsString("parsing_exception"), containsString("remote clusters are not supported in LOOKUP JOIN"))
                     : allOf(containsString("verification_exception"), containsString("Unknown index [foo]"))

